lib/hooks/useOSDetection.ts

```diff
@@ -0,0 +1,19 @@
+import { useEffect, useState } from "react";
+
+function useOSDetection(): boolean {
+  const [isMac, setIsMac] = useState<boolean>(false);
+
+  useEffect(() => {
+    // Constant pattern to test against
+    const macOSPatterns = /Mac|Macintosh|MacIntel|Mac OS X/;
+
+    // Initial check using userAgent
+    const userAgentCheck = macOSPatterns.test(window.navigator.userAgent);
```

Comment:
Prefer all caps for constans.

```suggestion
const MACOS_USER_AGENT_PART = /Mac|Macintosh|MacIntel|Mac OS X/;

function useOSDetection(): boolean {
  const [isMac, setIsMac] = useState<boolean>(false);

  useEffect(() => {

    // Initial check using userAgent
    const userAgentCheck = MACOS_USER_AGENT_PART.test(window.navigator.userAgent);
```

Reply:
Awesome thanks. Making the updates as we speak. For the styling, do you suggest we lower the font-size and increase padding? I tried and even to add `px-1` we would have to drop the font-size to `text-[10px]`. Feels a bit small.

Reply:
I'll let you know @isabensusan chime in.

Reply:
I can make it `text-[12px]` `px-1` and it looks pretty great. Would need to increase the width of the search box a bit to `w-56`.
<img width="280" alt="Screenshot 2023-09-29 at 23 24 12" src="https://github.com/open-sauced/app/assets/92145222/a9c39f58-caf4-4b42-818e-22721418d458">
<img width="1324" alt="Screenshot 2023-09-29 at 23 29 52" src="https://github.com/open-sauced/app/assets/92145222/b0ebee24-4221-4b2d-98c1-c6a6556b0c6f">

Reply:
@FrancoLab that seems reasonable to go with. We'll revisit if it needs adjustments in the future

Reply:
Thanks @brandonroberts . Will implement these changes 💪

Reply:
lgtm! thanks 😄

Reply:
I just realized in my refactor suggestions, that I forgot to rename the hook to, `function useIsMacOS()`. The code suggestion above has been updated @FrancoLab.

Reply:
Thanks everyone. New updates pushed. Ready for review.